resolwe/flow/models/utils.py

```diff
@@ -137,11 +141,14 @@ def validate_data(data_pk, type_):
                 "Data object of type `{}` is required, but type `{}` is given. "
                 "(id:{})".format(type_, data['process__type'], data_pk))
 
+    is_dirty = False
+    dirty_field = ''
     for _schema, _fields, _ in iterate_schema(instance, schema):
         name = _schema['name']
 
-        if test_required and _schema.get('required', True) and name not in _fields:
-            raise ValidationError("Required field \"{}\" not given.".format(name))
+        if not is_dirty and test_required and _schema.get('required', True) and name not in _fields:
+            is_dirty = True
+            dirty_field = name
```

Comment:
If multiple fields are dirty, this will just show the last one. Should we use a list instead?